extensions/replication/failedCRR/FailedCRRConsumer.js

```diff
@@ -87,7 +88,57 @@ class FailedCRRConsumer {
             log.end();
             return cb();
         }
-        return this._setRedisKey(data, kafkaEntry, log, cb);
+        return async.series([
+            next => this._removePreexistingKeys(data, err => {
+                if (err) {
+                    this.logger.error('error deleting preexisting CRR object ' +
+                    'metrics redis keys', {
+                        method: 'FailedCRRConsumer.processKafkaEntry',
+                        error: err,
+                    });
+                }
+                return next();
+            }),
+            next => this._setRedisKey(data, kafkaEntry, log, next),
+        ], cb);
+    }
+
+    /**
+     * Delete any pre-existing object CRR metrics keys. This way the object
+     * metrics during the retry will not be affected by the prior attempt.
+     * @param {Object} data - The kafka entry value
+     * @param {Function} cb - callback to call
+     * @return {undefined}
+     */
+    _removePreexistingKeys(data, cb) {
+        // eslint-disable-next-line no-unused-vars
+        const [service, extension, status, bucketName, objectKey, versionId,
+            site] = data.key.split(':');
+        // CRR object-level metrics key schema.
+        const queryString = `${site}:${bucketName}:${objectKey}:${versionId}:` +
+            `${service}:${extension}:*`;
+        return redisClient.batch([['keys', queryString]], (err, res) => {
```

Comment:
Using `keys *` pattern is a bad idea to use on a production Redis. Please use `scan` instead.
Reference from redis.io/commands/keys

> Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases.

Reply:
Good call, scan stream is handy.